sklearn/tests/test_dummy.py

```diff
@@ -102,6 +102,20 @@ def test_most_frequent_and_prior_strategy():
                                       clf.class_prior_.reshape((1, -1)) > 0.5)
 
 
+def test_most_frequent_and_prior_strategy_with_2d_column_y():
```

Comment:
Maybe add ref to this PR so we know why we need this non-regression test

Reply:
done!